src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -790,52 +834,128 @@ void Hole::updateDiameterParam()
     }
     else { // we have a clearance hole
         bool found = false;
-        int MatrixRowSize = sizeof(metricHoleDiameters) / sizeof(metricHoleDiameters[0]);
-        switch ( ThreadFit.getValue() ) {
-        case 0: /* standard fit */
-            // read diameter out of matrix
-            for (int i = 0; i < MatrixRowSize; i++) {
-                if (metricHoleDiameters[i][0] == diameter) {
-                    diameter = metricHoleDiameters[i][2];
-                    found = true;
-                    break;
+        std::string threadType = ThreadType.getValueAsString();
+        // UTS and metric have a different clearance hole set
+        if (threadType == "ISOMetricProfile" || threadType == "ISOMetricFineProfile") {
+            int MatrixRowSizeMetric = sizeof(metricHoleDiameters) / sizeof(metricHoleDiameters[0]);
+            switch (ThreadFit.getValue()) {
+            case 0: /* standard fit */
+                // read diameter out of matrix
+                for (int i = 0; i < MatrixRowSizeMetric; i++) {
+                    if (metricHoleDiameters[i][0] == diameter) {
+                        diameter = metricHoleDiameters[i][2];
+                        found = true;
+                        break;
+                    }
                 }
-            }
-            // if nothing was found (e.g. if not metric), we must calculate
-            if (!found) {
-                diameter = (5 * ((int)((diameter * 110) / 5))) / 100.0;
-            }
-            break;
-        case 1: /* close fit */
-            // read diameter out of matrix
-            for (int i = 0; i < MatrixRowSize; i++) {
-                if (metricHoleDiameters[i][0] == diameter) {
-                    diameter = metricHoleDiameters[i][1];
-                    found = true;
-                    break;
+                // if nothing was found (e.g. if not metric), we must calculate
+                if (!found) {
+                    diameter = diameter * 1.1;

```

Comment:
Having a helper function for this operation would reduce a lot of code repetition.

I.e  you could replace this with:

diameter = tableLookup(metricHoleDiameters, MatrixRowSizeMetric, diameter * 1.1);

Reply:
> Having a helper function for this operation would reduce a lot of code repetition.

I tried this now but did not gain shorter code. I still have to check 3 different cases for metric holes and UTS and the lookup is different for metric and UTS.

So if you have a clever proposal, please post it.

Reply:
At least this saves some repetion. And it will be useful in the future. 
https://github.com/donovaly/FreeCAD/pull/2

It it a pity it does not work for the UTS tables, as they use a struct instead of a pure double array.